framework/src/meshgenerators/BreakMeshByBlockGenerator.C

```diff
@@ -48,21 +52,44 @@ BreakMeshByBlockGenerator::validParams()
 BreakMeshByBlockGenerator::BreakMeshByBlockGenerator(const InputParameters & parameters)
   : BreakMeshByBlockGeneratorBase(parameters),
     _input(getMesh("input")),
-    _block(parameters.isParamSetByUser("block") ? getParam<std::vector<SubdomainID>>("block")
-                                                : std::vector<SubdomainID>()),
-    _block_set(_block.begin(), _block.end()),
-    _block_restricted(parameters.isParamSetByUser("block")),
+    _block_pairs_restricted(parameters.isParamSetByUser("block_pairs")),
+    _surrounding_blocks_restricted(parameters.isParamSetByUser("surrounding_blocks")),
     _add_transition_interface(getParam<bool>("add_transition_interface")),
     _split_transition_interface(getParam<bool>("split_transition_interface")),
-    _interface_transition_name(getParam<BoundaryName>("interface_transition_name"))
+    _interface_transition_name(getParam<BoundaryName>("interface_transition_name")),
+    _add_interface_on_two_sides(getParam<bool>("add_interface_on_two_sides"))
 {
+  if (_block_pairs_restricted && _surrounding_blocks_restricted)
+    paramError("'surrounding_blocks' and 'block_pairs' can not be used at the same time.");
+
+  if (_block_pairs_restricted)
+  {
+    for (auto block_pair : getParam<std::vector<std::vector<SubdomainID>>>("block_pairs"))
+    {
+      if (block_pair.size() != 2)
+        paramError("Each row of 'block_pairs' must have a size of two (block names).");
+      std::pair<SubdomainID, SubdomainID> pair =
+          std::make_pair(block_pair[0] < block_pair[1] ? block_pair[0] : block_pair[1],
+                         block_pair[0] < block_pair[1] ? block_pair[1] : block_pair[0]);
+      _block_pairs.insert(pair);
+      std::copy(block_pair.begin(), block_pair.end(), std::inserter(_block_set, _block_set.end()));
+    }
+  }
+  else if (_surrounding_blocks_restricted)
+    std::copy(getParam<std::vector<SubdomainID>>("surrounding_blocks").begin(),
+              getParam<std::vector<SubdomainID>>("surrounding_blocks").end(),
+              std::inserter(_block_set, _block_set.end()));
+
   if (typeid(_input).name() == typeid(DistributedMesh).name())
     mooseError("BreakMeshByBlockGenerator only works with ReplicatedMesh.");
 
   if (!_add_transition_interface && _split_transition_interface)
     paramError("split_transition_interface",
                "BreakMeshByBlockGenerator cannot split the transition interface because "
                "add_transition_interface is false");
+
+  if (_add_transition_interface && _block_pairs_restricted)
+    paramError("add_transition_interface cannot work with 'block_pairs'.");
```

Comment:
You need to have the parameter name as the first argument here!

Reply:
Got it.